selectrum.el

```diff
@@ -285,17 +285,29 @@ nothing to remove.)"
 
 (defun selectrum-default-candidate-preprocess-function (candidates)
   "Default value of `selectrum-preprocess-candidates-function'.
-Sort first by length and then alphabetically. CANDIDATES is a
-list of strings."
+Sort first by history position, then by length and then alphabetically.
+CANDIDATES is a list of strings."
   (if selectrum-should-sort
-      (sort candidates
-            (lambda (c1 c2)
-              (or (< (length c1)
-                     (length c2))
-                  (and (= (length c1)
-                          (length c2))
-                       (string-lessp c1 c2)))))
-    candidates))
+      (let* ((list (symbol-value minibuffer-history-variable))
```

Comment:
The history variable can also be set to `t` (when history should be ignored).

Reply:
I don't think this can happen - I copied this from the function `completion-all-sorted-completions` basically. Probably `read-from-minibuffer` replaces `t` by `minibuffer-history`?

Reply:
Hmm, checked again. The variable can be `t` according to the docstring. I wonder why `completion-all-soerted-completions` works then.